sys/aarch64/boot.c

```diff
@@ -199,6 +208,39 @@ __boot_text static paddr_t build_page_table(void) {
 
   l0[L0_INDEX(DMAP_BASE)] = (pde_t)l1d | L0_TABLE;
 
+#if KASAN /* Prepare KASAN shadow mappings */
+  kasan_sanitized_end =
+    SUPERPAGESIZE + roundup(va, SUPERPAGESIZE << KASAN_SHADOW_SCALE_SHIFT);
```

Comment:
`init_vm_page` does some weird assumptions.  Without that additional pages it's try to access not mapped pages in shadow map. This is also a reason why I am using `KASAN_MD_SANITIZED_END` instead of `_kasan_sanitized_end`.

Reply:
Hmm... `init_vm_page()` calls `vm_boot_alloc()`, which manually adjusts `vm_kernel_end` without calling `pmap_growkernel()` (which I guess it can't, since page allocation doesn't work yet), so I think it's possible for  `vm_kernel_end` to go past `_kasan_sanitized_end`. Maybe we're enabling KASAN a bit too early?

Reply:
Solving this problem is outside the scope of this PR.